docs/docs/noir/standard_library/append.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Append Trait
+description:
+  The Append Trait abstracts over types that can be appended to
+keywords: [append, trait]
+---
+
+`Append` can abstract over types that can be appended to - usually container types:
+
+```rs
+trait Append {
+    fn empty() -> Self;
+
+    fn append(self, other: Self) -> Self;
+}
+```
+
+`Append` requires two methods:
+
+- `empty`: Constructs an empty value of `Self`.
+- `append`: Append two values together, returning the result.
+
+Additionally, it is expected that for any implementation:
+
+- `T::empty().append(x) == x`
+- `x.append(T::empty()) == x`
+
+---
+
+## Implementations
```

Comment:
Do we regularly add docs for each impl? If not, this may get out of date and users may assume that the list is complete.

Reply:
Yes, see https://66883a76f28330a67a4baa9b--noir-docs.netlify.app/docs/dev/noir/standard_library/traits/.

I missed that file before, `Append` should probably be added there rather than in a new file.

We should really reorganize the docs sometime, there's really no organization to them.